addon/index.js

```diff
@@ -109,13 +109,15 @@ class EmberRouterScroll extends EmberRouter {
 
     if (!scrollWhenIdle && !scrollWhenAfterRender) {
       // out of the option, this happens on the tightest schedule
-      scheduleOnce('render', this, CALLBACK.bind(this, transition));
-    } else if (scrollWhenAfterRender) {
+      scheduleOnce('render', this, CALLBACK, transition);
+    } else if (scrollWhenAfterRender && !scrollWhenIdle) {
       // out of the option, this happens on the tightest schedule
-      scheduleOnce('afterRender', this, CALLBACK.bind(this, transition));
+      scheduleOnce('afterRender', this, CALLBACK, transition);
     } else {
       // as described in ember-app-scheduler, this addon can be used to delay rendering until after the route is idle
-      whenRouteIdle().then(CALLBACK.bind(this, transition));
+      whenRouteIdle().then(() => {
```

Comment:
Need some sort of deferred queue that allows this case to be updated with latest